apps/admin-gui/src/app/shared/components/dialogs/applications-list-columns-change-dialog/applications-list-columns-change-dialog.component.ts

```diff
@@ -74,32 +100,74 @@ export class ApplicationsListColumnsChangeDialogComponent implements OnInit {
               this.attribute = attribute;
               const configuredColumns: string[] = attribute.value as string[];
               if (configuredColumns !== null && configuredColumns.length > 0) {
-                this.selectedColumns.setValue(configuredColumns);
+                this.selectedColumns = this.friendlyToDisplayAttrNames(configuredColumns);
               } else {
-                this.selectedColumns.setValue(this.simpleColumns);
+                this.selectedColumns = this.simpleColumns;
               }
             });
+          this.loading = false;
         });
     }
   }
 
   confirm(): void {
+    if (
+      this.selectedColumns === null ||
+      this.selectedColumns.length === 0 ||
+      containsExactlyInAnyOrder(this.selectedColumns, this.simpleColumns)
+    ) {
+      this.attribute.value = [];
+    }
     if (this.data.groupId) {
       this.changeGroupAttribute();
     } else {
       this.changeVoAttribute();
     }
   }
+
   cancel(): void {
     this.dialogRef.close(false);
   }
   default(): void {
-    this.attribute.value = [];
-    this.confirm();
+    this.selectedColumns = [...this.simpleColumns];
+  }
+
+  template(): void {
+    this.selectedColumns = this.friendlyToDisplayAttrNames(
+      this.templateAttribute.value as string[]
+    );
+  }
+
+  toggleEvent(names: string[]): void {
+    // Replace array in-place so it won't trigger ngOnChanges
+    this.selectedColumns.splice(0, this.selectedColumns.length, ...names);
+  }
+
+  refreshOnClosed(): void {
```

Comment:
is this function necessary? what does it do?

Reply:
copied it from similar component, probably redundant. removed